### ISSUE TITLE
Can not serialize CompletionStage<Response>

### ISSUE THREAD

**Issue by hantsy** [eclipse-ee4j/jersey#5103]
Glassfish 7.0.0-M7
Java 17
Windows 10

Sample Jaxrs resource return a `CompletionStage<Response>`.

```java
@GET
public CompletionStage<Response> getAllTodos() {
    return todoService.getAllTodosAsync().thenApply(todos -> Response.ok(todos).build());
}
```
When running the test [TodoResourceTest](https://github.com/hantsy/jakartaee10-sandbox/blob/master/concurrency/src/test/java/com/example/it/TodoResourceTest.java), and got the following exceptions in the server.log.

```bash
[2022-07-20T23:09:41.977+0800] [glassfish 7.0] [WARNING] [] [org.glassfish.jersey.server.ServerRuntime$Responder] [tid: _ThreadID=139 _ThreadName=java:module/concurrent/MyExecutor-managedThreadFactory-Thread-1] [timeMillis: 1658329781977] [levelValue: 900] [[
  An exception mapping did not successfully produce and processed a response. Logging the exception propagated to the default exception mapper.
jakarta.ws.rs.ProcessingException: Error writing JSON-B serialized object.
	at org.glassfish.jersey.jsonb.internal.JsonBindingProvider.writeTo(JsonBindingProvider.java:105)
```

Attached the completed [server.log](https://github.com/eclipse-ee4j/jersey/files/9151511/server.log)

**Comment by jansupol** [comment]
> Caused by: jakarta.json.bind.JsonbException: Unable to serialize property 'allowedMethods' from java.util.Vector
	at org.eclipse.yasson.internal.serializer.ObjectSerializer.lambda$serialize$0(ObjectSerializer.java:43)
	at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:721)
	at org.eclipse.yasson.internal.serializer.ObjectSerializer.serialize(ObjectSerializer.java:38)
	at org.eclipse.yasson.internal.serializer.RecursionChecker.serialize(RecursionChecker.java:38)
	at org.eclipse.yasson.internal.serializer.KeyWriter.serialize(KeyWriter.java:41)
	at org.eclipse.yasson.internal.serializer.NullVisibilitySwitcher.serialize(NullVisibilitySwitcher.java:40)
	at org.eclipse.yasson.internal.serializer.NullSerializer.serialize(NullSerializer.java:67)
	at org.eclipse.yasson.internal.SerializationContextImpl.serializeObject(SerializationContextImpl.java:197)
	at org.eclipse.yasson.internal.SerializationContextImpl.marshall(SerializationContextImpl.java:133)
	at org.eclipse.yasson.internal.SerializationContextImpl.marshallWithoutClose(SerializationContextImpl.java:170)
	at org.eclipse.yasson.internal.JsonBinding.toJson(JsonBinding.java:142)
	at org.glassfish.jersey.jsonb.internal.JsonBindingProvider.writeTo(JsonBindingProvider.java:103)
	... 32 more
Caused by: jakarta.json.bind.JsonbException: Error getting value on: [Todo{id=59eb3441-b774-4c4a-b1c6-c080fa0e2ea7, title='What's new in JPA 3.1?', completed=false}, Todo{id=e9fdd67f-c90d-4c1a-9f9b-05df8805bfb4, title='What's new in Jaxrs 3.1', completed=false}, Todo{id=160b83d3-5985-4063-a9df-cb1865fdf064, title='Learn new features in Faces 4.0', completed=false}]
	at org.eclipse.yasson.internal.serializer.ValueGetterSerializer.serialize(ValueGetterSerializer.java:41)
	at org.eclipse.yasson.internal.serializer.ObjectSerializer.lambda$serialize$0(ObjectSerializer.java:41)
	... 43 more
Caused by: java.lang.ClassCastException: Cannot cast java.util.Vector to jakarta.ws.rs.core.Response
	at java.base/java.lang.Class.cast(Class.java:3889)
	at org.eclipse.yasson.internal.serializer.ValueGetterSerializer.serialize(ValueGetterSerializer.java:39)
	... 44 more

This looks more like a [Yasson ](https://github.com/eclipse-ee4j/yasson/issues) issue.

**Comment by jansupol** [comment]
The thing is that Jersey does not handle `CompletableFuture<Response>`, but it can handle `CompletableFuture<List<Todo>>`. If you change this, it will be working fine. Your example uses functionality not supported by Jersey, and Yasson has a bug in the error message, it should say _Unable to serialize property 'allowedMethods' from Response_.

**Comment by hantsy** [comment]
Jsonb impl(Yasson) provides basic serialization for general purpose.

As a Jaxrs impl, to support Async using Java 8 `CompletionStage/CompletableFuture`, Jersey should add its own serializer/desearializer for supporting `CompletableFuture<Response>`.

As you mentioned, to get a list(`GET 200` status), use `CompletableFuture<List<Todo>>` is ok, how to handle `201 Created with created uri as locaiton header` status and `204 no Content`? In the async handling, I would like use `CompletableFuture<Response>` in all cases.

Quarkus supports such an approach in the early version, check my example [CompletableFuture<Response>](https://github.com/hantsy/quarkus-sandbox/blob/master/legacy/java8/src/main/java/com/example/PostResource.java), in the new version, it is replaced by RactiveStreams(Small Mutiny).

In contrast to Spring, besides the async `Completable<ResponseEntity>`, it supports Spring WebFlux reactive `Mono<ResponseEntity>` and `ResponseEntity<Mono or Flux>` as return type. 

So I think to support async in Jaxrs, it should add it own serializer to handle `CompletableFuture<Response>` and `Response<CompetableFuture>`,  and for all JSON implementation providers(jsonb, jackson, json-p).

**Comment by jansupol** [comment]
`CompletionStage<Response>` is fine to implement, Till then, for HTTP status other than 200, you can use `AsyncResponse`.

**Comment by hantsy** [comment]
It is not included in the Glassfish 7.0.0-M9?
